openfisca_core/variables.py

```diff
@@ -237,37 +252,92 @@ def set_reference(self, reference):
 
     def set_base_function(self, base_function):
         if not base_function and self.baseline_variable:
-            return self.baseline_variable.formula.base_function.im_func
-        if self.definition_period == ETERNITY:
-            if base_function and base_function not in [permanent_default_value, requested_period_default_value]:
-                raise ValueError('Unexpected base_function {}'.format(base_function))
-            return requested_period_default_value
-
-        if self.is_period_size_independent:
-            if base_function is None:
-                return requested_period_last_value
-            if base_function in [missing_value, requested_period_last_value, requested_period_last_or_next_value]:
-                return base_function
-            raise ValueError('Unexpected base_function {}'.format(base_function))
-
-        if base_function is None:
-            return requested_period_default_value
+            return self.baseline_variable.base_function
+
+        if base_function and base_function not in {
+                missing_value,
+                requested_period_default_value,
+                requested_period_last_or_next_value,
+                requested_period_last_value
+                }:
+            raise ValueError(u'Unexpected base_function {}'.format(base_function).encode('utf-8'))
+
+        if self.is_period_size_independent and base_function is None:
+            return requested_period_last_value
 
         return base_function
 
     def set_set_input(self, set_input):
         if not set_input and self.baseline_variable:
-            return self.baseline_variable.formula.set_input.im_func
+            return self.baseline_variable.set_input
         return set_input
 
     def set_calculate_output(self, calculate_output):
         if not calculate_output and self.baseline_variable:
-            return self.baseline_variable.formula.calculate_output.im_func
+            return self.baseline_variable.calculate_output
         return calculate_output
 
+    def set_formulas(self, formulas_attr):
+        formulas = SortedDict()
+        for formula_name, formula in formulas_attr.iteritems():
```

Comment:
iteritems() is not compatible with python3 :(
I think we can use .items() instead